docs/reference/transactions.md

```diff
@@ -81,9 +82,9 @@ Includes all fields in [Header](#common-fields-header-and-type) and `"type"` is
 
 |Field|Required|Type|codec| Description|
 |---|---|---|---|---|
+|<a name="sender">Sender</a>| _required_ |Address|`"snd"`|The address of the account that signs, pays the fee and sends the asset.|
 |<a name="xferasset">XferAsset</a>| _required_ |uint64|`"xaid"`|The unique ID of the asset to be transferred.|
 |<a name="assetamount">AssetAmount</a>|_required_|uint64|`"aamt"`| The amount of the asset to be transferred. A zero amount transferred to self allocates that asset in the account's Asset map.|
-|<a name="assetsender">AssetSender</a>|_required_|Address|`"asnd"`|The sender of the transfer. The regular [sender](#sender) field should be used and this one set to the zero value for regular transfers between accounts. If this value is nonzero, it indicates a clawback transaction where the [sender](#sender) is the asset's clawback address and the asset sender is the address from which the funds will be withdrawn.|
```

Comment:
I don't know if a transaction can be submitted to the chain without `asnd` being set to zero though. So to completely delete this field would also mislead people as to how submit a standard transaction. If leaving it blank is an implicit zero then I think leaving it out of the docs would be okay.

Reply:
From what I've tested, it actually must not be set (the key must not be present). Using `null` (or `0` for numbers) causes the signing to fail. See also https://github.com/algorand/docs/pull/454

Reply:
Ahhh that makes sense, good catch! That's right it's an address not a boolean field! Yeah that should be reworded as well. Okay then yeah I think this edit should be fine.

Reply:
Hold up, I think what they may be referring to is something called the "Zero Address" which I don't think equates to the integer value of 0
![CleanShot 2021-07-02 at 20 35 54](https://user-images.githubusercontent.com/11368049/124338334-49c16600-db75-11eb-8c6a-fec60803d928.png)

Reply:
Exactly. When it comes to `AssetTransfer` transaction type attention must be paid to the difference between the `Sender` and `AssetSender`. When an `AssetTransfer` transaction is executed as `AssetClawback`,  then the `Clawback Address` is the `Sender` (who initiate and authorize the transaction) while the `AssetSender` (normally set to `ZeroGlobalAddress`, which is not a boolean value) is the address of the account form which the `AssetAmount` is taken form. So I think all the definition of `Sender` and `AssetSender` for the `AssetTransfer` should give clear evidence of this principle.

Reply:
I've the impression that the signature validator has a strict mapping policy, which for "zero values" is that the key is not existent.

But I can double check: what is the zero address? I did a test with an address with the 32 bytes set to 0 and the validation failed. As mentioned previously, a `null` value also makes it fail. `0` was in reference to a numeric field: https://github.com/algorand/docs/pull/454